src/main/java/org/jabref/logic/autosaveandbackup/BackupManager.java

```diff
@@ -189,7 +198,8 @@ private Optional<Path> determineBackupPathForNewBackup() {
      * @param backupPath the path where the library should be backed up to
      */
     private void performBackup(Path backupPath) {
-        if (!needsBackup) {
+
+        if (!needsBackup || discardedFileExists) {
```

Comment:
Why the check here? A backup should be performed regardless of the discared file. The discarded file was of the **last** JabRef run.